app/controllers/ckey2bibframes_controller.rb

```diff
@@ -0,0 +1,29 @@
+# Controller for the Ckey2bibframe form
+class Ckey2bibframesController < ApplicationController
+  def new
+    @ckey2bibframe = Ckey2bibframe.new
+  end
+
+  def create
+    @ckey2bibframe = Ckey2bibframe.new(ckey2bibframe_params)
+
+    if @ckey2bibframe.valid?
+      render action: 'show'
+    else
+      flash[:warning] = 'Check that all form fields are entered!'
+      render action: 'new'
+    end
+  end
+
+  def show
+    params[:baseuri] ||= 'http://ld4p.stanford.edu'
+    @ckey2bibframe = Ckey2bibframe.new(baseuri: params[:baseuri], ckey: params[:ckey])
```

Comment:
Good idea. That default uri appears twice in the code. I will add it to the settings.